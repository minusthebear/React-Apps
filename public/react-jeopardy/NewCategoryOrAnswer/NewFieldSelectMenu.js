import React from 'react';
import Select from "react-select";

const NewFieldSelectMenu = ({testId, options, onChange}) => {
    return (
        <div data-testid={testId} className={ 'newFieldSelectMenu-' + testId } style={{width: '200px', textAlign:'center', margin: '0 auto'}}>
            <Select options={options} onChange={onChange}/>
        </div>
    );
};

export default NewFieldSelectMenu;