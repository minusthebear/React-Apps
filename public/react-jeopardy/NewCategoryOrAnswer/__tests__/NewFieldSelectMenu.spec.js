import React from 'react';
import {shallow, mount} from 'enzyme';
import NewFieldSelectMenu from "../NewFieldSelectMenu";
import renderer from 'react-test-renderer';

// NOTE: if jest.mock('react-select') is not at top of page, many unit tests will fail
jest.mock('react-select', () => ({options, onChange}) =>
    <select>Select</select>
);

describe('NewFieldSelectMenu', () => {
    it('smoke test', () => {
        const wrapper = shallow(<NewFieldSelectMenu onChange={() => {}} options={[]} testId={'smokeTest'} />);
    });

    it('.newFieldSelectMenu div', () => {
        const wrapper = shallow(<NewFieldSelectMenu onChange={() => {}} options={[]} testId={'msgInABottle'} />);
        const div = wrapper.find('.newFieldSelectMenu-msgInABottle');
        expect(div.length).toBe(1);
    });

    it('..newFieldSelectMenu child select', () => {
        const wrapper = mount(<NewFieldSelectMenu onChange={() => {}} options={[]} testId={'msgInABottle'} />);
        const select = wrapper.find('select');
        expect(select.length).toBe(1);
    });
});

describe("make snapshot", () => {

    it('renders as expected', () => {
        const tree = renderer.create(<NewFieldSelectMenu testId={'smokeTest'} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});