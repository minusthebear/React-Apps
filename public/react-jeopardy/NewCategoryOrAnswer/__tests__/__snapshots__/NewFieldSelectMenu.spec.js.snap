// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot renders as expected 1`] = `
<div
  className="newFieldSelectMenu-smokeTest"
  data-testid="smokeTest"
  style={
    Object {
      "margin": "0 auto",
      "textAlign": "center",
      "width": "200px",
    }
  }
>
  <select>
    Select
  </select>
</div>
`;
